src/library.js

```diff
@@ -642,11 +642,30 @@ LibraryManager.library = {
 
   system__deps: ['__setErrNo'],
   system: function(command) {
+#if ENVIRONMENT_MAY_BE_NODE
+    if (ENVIRONMENT_IS_NODE) {
+      if (!command)
+        return 1; // shell is available
+
+      let cp = require('child_process');
+      // this may fail if the stdout of program is too big...
+      // sadly there is no option for spawnSync to ignore stdout
+      // https://nodejs.org/api/child_process.html#child_process_child_process_spawnsync_command_args_options
+      let ret = cp.spawnSync(UTF8ToString(command), [], {shell:true, stdio:'ignore'});
+
+      if (ret.status === null) {
+        ___setErrNo({{{ cDefine('EAGAIN') }}});
+        return -1;
+      }
+
+      return ret.status;
+    }
+#endif // ENVIRONMENT_MAY_BE_NODE
     // int system(const char *command);
     // http://pubs.opengroup.org/onlinepubs/000095399/functions/system.html
     // Can't call external programs.
     ___setErrNo({{{ cDefine('EAGAIN') }}});
-    return -1;
+    return (command ? -1 : 0 /* no shell available */);
```

Comment:
I think this might be a little clearer (just opinion, feel free to disagree):
```
[...]
		let ret = cp.spawnSync(UTF8ToString(command), [], {shell:true, stdio:'ignore'});
		if (ret.status !== null) {
			return ret.status;
		}
	}
#endif // ENVIRONMENT_MAY_BE_NODE
	___setErrNo({{{ cDefine('EAGAIN') }}});
	return (-1);
},
```


Then you don't have the duplicate `___setErrNo({{{ cDefine('EAGAIN') }}});`

Reply:
How is this gonna work when you return from the function?

Reply:
Note I dropped the EAGAIN in node code because it didn't match the real behaviour.